Src/FluentAssertions/Equivalency/AssertionRule.cs

```diff
@@ -45,20 +45,25 @@ public bool AssertEquality(IEquivalencyValidationContext context)
         {
             if (predicate(context))
             {
-                bool expectationisNull = ReferenceEquals(context.Expectation, null);
+                bool subjectSucceeded =
```

Comment:
I had a hard time understanding the algorithm here. I suspect it's in the naming of things. For instance, both `subjectSucceeded` and `expectationSucceeded ` doesn't say much about what the assertion is trying to do. Same for the other changes in this PR.

Reply:
These two commits helped me in trying to understand what was going on:
92fd4e10a7c77201ff2006b41ed1d7e01f384013
7914d73769e9bd6562022215456773eec55aebd6